org.civicrm.reqangular/src/common/decorators/angular-date/datepicker-addon.js

```diff
@@ -1,35 +1,88 @@
-define([], function () {
+/* eslint-env amd */
+
+define([
+  'common/moment'
+], function (moment) {
   'use strict';
 
   /**
    * Decorates the uib-datepicker-popup directive, making its sibling '.input-group-addon'
    * trigger the datepicker
    */
-  return ['$delegate', function ($delegate) {
+  return ['$delegate', '$compile', function ($delegate, $compile) {
     var directive = $delegate[0];
+    var isMobile = document.getElementsByTagName('body')[0].classList.contains('mobile');
     var origCompile = directive.compile;
+    var calendarIcon = '<span ' +
+      'class="input-group-addon pointer">' +
+      '<i class="fa fa-calendar"></i>' +
+      '</span>';
 
-    directive.compile = function compile() {
-      var link = origCompile.apply(this, arguments);
+    if (isMobile) {
```

Comment:
too much stuff going on in this `if`, put it into a separate function

Reply:
Organised the code.